### ISSUE TITLE
[vue/no-ref-as-operand] Not working as expected

### ISSUE THREAD

**Issue by ThibaultVlacich** [vuejs/eslint-plugin-vue#3128]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 10.8.1
- **eslint-plugin-vue version:** 10.10.0
- **Vue version:** 3.5.41
- **Node version:** N/A
- **Operating System:** N/A 

**Please show your full configuration:**
```js
import pluginVue from 'eslint-plugin-vue';
import globals from 'globals';

export default [
  // add more generic rulesets here, such as:
  // js.configs.recommended,
  ...pluginVue.configs['flat/recommended'],
  // ...pluginVue.configs['flat/vue2-recommended'], // Use this if you are using Vue.js 2.x.
  {
    rules: {
      // override/add rules settings here, such as:
      // 'vue/no-unused-vars': 'error'
    },
    languageOptions: {
      sourceType: 'module',
      globals: {
        ...globals.browser,
      },
    },
  },
];

```

**What did you do?**	
Use a ref / computed without `.value` that was declared in a composable.

Cf reproduction Stackblitz for details.

**What did you expect to happen?**
Rule warns me that I forgot to use `.value` when using the ref

**What actually happened?**
Rule did not warn me.

**Repository to reproduce this issue**
https://stackblitz.com/edit/vitejs-vite-1davhtxy?file=src%2Fcomponents%2FTestStatus.vue

**Comment by ValentinYoushkevich** [comment]
Looked at the repro. `status` comes from `useStatus()` defined in another file, and the check is on a property, `if (status.isEnabled)`. The rule only knows about refs created with `ref()`, `computed()` and so on in the file being linted, and it only checks plain identifiers, so it can't see this one.

Since ESLint looks at one file at a time, catching refs returned from composables needs type information, which is what #2519 is about. That would require TypeScript with typed linting, so in a plain JS project like the repro it isn't really possible.

**Comment by FloEdelmann** [comment]
Closing in favor of #2519.
